### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by PhilippeFerreiraDeSousa**
- Package JLC2JSON can be installed and used again as a binary
- 3d Models are not jsons but normal text and them were json serialized before writing to disk
- make sleep random of expected value 1 second before each component query to easyeda.com
- do not pretty print jsons on disk to save space
- `response` was out of scope in except blocks
- There can technically be several 3d models and symbols, do not stop at the first one (I have not see any example of this happening though)
- do not delete log file at each run, put in place rotating logs each hour
- deduplicated code
- split into 3 files

### CODE REVIEW

**Review by bhusang — APPROVED** [review]
Wow, your code is very nice! Thank you for fixing/improving the tool.

File: JLC2JSON/JLC2JSON.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    args = parser.parse_args()
+    set_logging(args.logging_level, args.log_file)
+
+    for dir in (JSON_COMPONENT_DIR, JSON_SYMBOL_DIR, JSON_FOOTPRINT_DIR, JSON_3D_DIR):
+        create_dir(dir)
+
+    lcsc_id_file_path = args.components[0]
+    with open(lcsc_id_file_path, "rt") as file:
+        for lcsc_id in file.readlines():
```
Comment (ptuckey) [inline JLC2JSON/JLC2JSON.py:50]: Maybe log something every N lines, so we can see where it's at while it's running?
```
0 of 200,000...
1000 of 200,000...
...
```

**Review by ptuckey — APPROVED** [review]
Nicely done!!

Some open questions, which you may be addressing as follow-ups anyway...

1. S3 reading/writing? (you already mentioned you're on it)
2. Can we tail logs from AWS Lambda instances, or do we have to send them somewhere?
3. For parallization, is the idea to just install/fire this off on separate machines?

**Comment by ptuckey** [comment]
One other thought....  it's going to be hard to merge in any updates from the upstream repo now (which appears under active development).

Probably fine though, as we're customizing it for our own needs!

**Comment by bhusang** [comment]
I think that given the new tool does not rely on any code from the upstream repo, we should move the tool to a new repo entirely or even our backend tools.

File: JLC2JSON/JLC2JSON.py (hunk 1 above)
Comment (PhilippeFerreiraDeSousa) [reply JLC2JSON/JLC2JSON.py:50]: It does print everything it does but without index count

Comment (ptuckey) [reply JLC2JSON/JLC2JSON.py:50]: Can we add an index that prints out every 10 seconds or so?

I anticipate that we'll want to tail the log every now and then, and know how far we are, at a glance.

**Pull request merged**